torchdata/datapipes/iter/load/iopath.py

```diff
@@ -183,8 +183,12 @@ def __init__(
     def __iter__(self) -> Iterator[str]:
         for meta, data in self.source_datapipe:
             filepath = meta if self.filepath_fn is None else self.filepath_fn(meta)
-            with self.pathmgr.open(filepath, self.mode) as f:
-                f.write(data)
+
+            with iopath.file_lock(filepath):
```

Comment:
This introduces dependency on iopath, whereas iopath itself depends on portalocker for file locks (https://github.com/facebookresearch/iopath/blob/4481d29d6ae873a2413de5b1aba41c0b0a723f75/iopath/common/file_io.py#L98).may be we should instead use portalocker directly?

Reply:
That's an easy change, I saw that `iopath` was already used in the file which is why I picked it. Curious why you think it'd be better to use portalocker? Just about reducing a layer of indirection?

Reply:
Yes, reducing a layer of indirection is one reason. The other being is I think we no longer want to have file_locks as optional dependency since we are adding it to core datapipe. iopath as of now is optional and is needed only when users work with iopath related datapipes.